### PR TITLE
NXP-30050: write generic message for exception mapped to HTTP error code >= 500

### PR DESCRIPTION
**Pull request by troger**

### CODE REVIEW

File: modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java [hunk 1]
```diff
@@ -102,6 +104,19 @@ protected static void writeException(JsonGenerator jg, Throwable t, MediaType me
...
+     * @since 11.5
+     */
+    protected static String getExceptionMessage(String exceptionMessage, int statusCode) {
+        if (statusCode < SC_INTERNAL_SERVER_ERROR) {
+            return exceptionMessage;
+        }
+
+        boolean stackDisplay = jsonFactoryManager.isStackDisplay();
+        boolean isDevMode = Framework.isDevModeSet();
+        return stackDisplay || isDevMode ? exceptionMessage : "Internal Server Error";
```
Comment (efge) [inline modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java:117]: Why not simply

    if (statusCode < SC_INTERNAL_SERVER_ERROR || jsonFactoryManager.isStackDisplay() || Framework.isDevModeSet()) {
        return exceptionMessage;
    } else {
        return "Internal Server Error";
    }

**Review by efge — DISMISSED** [review]

**Review by efge — APPROVED** [review]

File: modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java [hunk 2]
```diff
@@ -102,6 +104,17 @@ protected static void writeException(JsonGenerator jg, Throwable t, MediaType me
         jg.flush();
     }
 
+    /**
+     * @since 11.5
+     */
+    protected static String getExceptionMessage(String exceptionMessage, int statusCode) {
+        if (statusCode < SC_INTERNAL_SERVER_ERROR || jsonFactoryManager.isStackDisplay() || Framework.isDevModeSet()) {
```
Comment (BoboTiG) [inline modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java:111]: Maybe a stupid question: what about 502, 503, 504 errors? They wil have the same `"Internal Server Error"` message?

Comment (BoboTiG) [reply modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java:111]: Well, it is the desired behavior. :)

**Review by kevinleturc — APPROVED** [review]

File: modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java (hunk 2 above)
Comment (troger) [reply modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java:111]: So, no question? 😬

Comment (efge) [reply modules/platform/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/JsonWebengineWriter.java:111]: Indeed we could try to keep better specific messages for >500 but the http code is what's important to clients. And there's very few places in Nuxeo where we return something else than 500.

**Review by efge — APPROVED** [review]

**Review by kevinleturc — APPROVED** [review]

**Pull request merged**